### PR TITLE
Fix issue with middleware and minor bugs in postrouter

### PR DESCRIPTION
**Pull request by plausibly**
- db.User was used in postcontroller instead of db.Post
- The !offset check was preventing offset = 0
- Updated auth to properly set payload of tokens (it was previously not working as intended)
- Updated post controller to accept capacity of 0
- Auth middleware is called before post route

### CODE REVIEW

**Review by danielduong8 — APPROVED** [review]

File: server/routes/v1/index.ts [hunk 1]
```diff
@@ -1,10 +1,12 @@
...
 import userRouter from './user';
 import postRouter from './post';
+import auth from '../../middleware/auth';
+import db from '../../models';
 
 const routes = express.Router();
 
 routes.use('/users/', userRouter);
-routes.use('/posts/', postRouter);
+routes.use('/posts/', auth(db.User), postRouter);
```
Comment (andy9775) [inline server/routes/v1/index.ts:10]: See #42 for a more through implementation.

File: server/routes/v1/post.ts [hunk 2]
```diff
@@ -9,12 +9,12 @@ const postController = new PostController(db.User);
 postRouter.get('', async (req: Request, res: Response) => {
   const limit = req.body.limit;
   const offset = req.body.offset;
-  if (!limit || !offset) {
+  if (!limit || (!offset && offset != 0)) {
```
Comment (andy9775) [inline server/routes/v1/post.ts:12]: A value of 0 for the offset is valid. Should check whether it is defined.

**Review by andy9775 — CHANGES_REQUESTED** [review]

File: server/controllers/v1/user.ts [hunk 3]
```diff
@@ -56,14 +59,15 @@ export default class UserController {
...
+          data: { message: "Email has not been confirmed" },
+        };
       }
 
       this.updateLastLogin(userWithPass);
 
-      const user = Object.assign({}, userWithPass);
-
-      delete user.password;
+      const user = { userName: userWithPass.userName, id: userWithPass.id };
```
[the remark this answers is not in the document]
Comment (plausibly) [reply server/controllers/v1/user.ts:70]: Object.assign does not work as intended, it returns nothing.

File: server/controllers/v1/post.ts [hunk 4]
```diff
@@ -141,7 +141,7 @@ export default class PostController {
     location?: string,
     capacity?: number
   ): Promise<{status: number; data: {result?: Post; message?: string}}> {
-    if (!title || !body || !location || !capacity) {
+    if (!title || !body || !location || (!capacity && capacity !== 0)) {
```
[the remark this answers is not in the document]
Comment (plausibly) [reply server/controllers/v1/post.ts:144]: Capacity 0 does not return an error here, we are checking `!capacity` and then check to make that wasn't trigerred from capacity being zero

`!capacity && capacity !== 0`

**Pull request closed without merging**